### PR TITLE
Add documentation re alphabetical sorted of MV dimensions

### PR DESCRIPTION
**Pull request by lkm**
Simple updating of documentation. It's very relevant to know that that multi-value dimensions are sorted by default and that it's possible to change this behaviour. 

To someone more intune with Druid's inner workings: Do we really want alphabetical sorted to be the default for MV dimensions? Would it not be better to default to `ARRAY` in DimensionSchema L89?

#### Added a forbidden-apis entry ...

<hr>

This PR has:
- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

<hr>

##### Key changed/added classes in this PR
 * `MyFoo`
 * `OurBar`
 * `TheirBaz`

### CODE REVIEW

File: docs/ingestion/index.md [hunk 1]
```diff
@@ -480,6 +480,7 @@ Dimension objects can have the following components:
 | type | Either `string`, `long`, `float`, or `double`. | `string` |
 | name | The name of the dimension. This will be used as the field name to read from input records, as well as the column name stored in generated segments.<br><br>Note that you can use a [`transformSpec`](#transformspec) if you want to rename columns during ingestion time. | none (required) |
 | createBitmapIndex | For `string` typed dimensions, whether or not bitmap indexes should be created for the column in generated segments. Creating a bitmap index requires more storage, but speeds up certain kinds of filtering (especially equality and prefix filtering). Only supported for `string` typed dimensions. | `true` |
+| multiValueHandling | Specify the type of handling multi-value fields will get, possible values are `SORTED_ARRAY`, `SORTED_SET` and `ARRAY`. The first two will order the array upon ingestion and `SORTED_SET` will remove duplicates. `ARRAY` will ingest data as-is | `SORTED_ARRAY` |
```
Comment (techdocsmith) [inline docs/ingestion/index.md:483]:
```suggestion
| multiValueHandling | Specify the type of handling for multi-value fields. Possible values are `SORTED_ARRAY`, `SORTED_SET`, and `ARRAY`. `SORTED_ARRAY` and `SORTED_SET` order the array upon ingestion and `SORTED_SET` removes duplicates. `ARRAY` ingests data as-is | `SORTED_ARRAY` |
```

File: docs/querying/multi-value-dimensions.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ title: "Multi-value dimensions"
 
 Apache Druid supports "multi-value" string dimensions. These are generated when an input field contains an
 array of values instead of a single value (e.g. JSON arrays, or a TSV field containing one or more `listDelimiter`
-characters).
+characters), by default they are sorted alphabetically, see [Dimension Objects](../ingestion/index.html#dimension-objects) for configuration.
```
Comment (techdocsmith) [inline docs/querying/multi-value-dimensions.md:28]:
```suggestion
characters), by default Druid lists the values in alphabetical order, see [Dimension Objects](../ingestion/index.md#dimension-objects) for configuration.
```
Please use `.md` for links. Also, it's not clear to me what the related information is at the destination. But I'm a still fairly new to Druid.

**Review by techdocsmith — CHANGES_REQUESTED** [review]
See comments. Please fix link style.

File: docs/querying/multi-value-dimensions.md (hunk 2 above)
Comment (lkm) [reply docs/querying/multi-value-dimensions.md]: Thanks @techdocsmith, I've updated the changes. "Lists" is the wrong verb here, "ingests" would be more appropriate. The referred section is the above committed detail about multiValueHandling.

Comment (lkm) [reply docs/querying/multi-value-dimensions.md]: Also `.md` should be `.html`, check the rest of the reference in the file. The markdown is compiled for the web version of the doc.

Comment (techdocsmith) [reply docs/querying/multi-value-dimensions.md]: @lkm , please see: https://github.com/apache/druid/pull/10528 Using .md is more flexible for viewing docs locally and for publishing to the web

Comment (lkm) [reply docs/querying/multi-value-dimensions.md]: Thanks for the context @techdocsmith, I've changed it to .md

File: docs/querying/multi-value-dimensions.md [hunk 3]
```diff
@@ -25,7 +25,7 @@ title: "Multi-value dimensions"
 
 Apache Druid supports "multi-value" string dimensions. These are generated when an input field contains an
 array of values instead of a single value (e.g. JSON arrays, or a TSV field containing one or more `listDelimiter`
-characters).
+characters), by default Druid ingests the values in alphabetical order, see [Dimension Objects](../ingestion/index.md#dimension-objects) for configuration.
```
Comment (gianm) [inline docs/querying/multi-value-dimensions.md:28]: This is a run on sentence. Please replace the comma with a period and split up the sentences.

File: docs/ingestion/index.md [hunk 4]
```diff
@@ -480,6 +480,7 @@ Dimension objects can have the following components:
 | type | Either `string`, `long`, `float`, or `double`. | `string` |
 | name | The name of the dimension. This will be used as the field name to read from input records, as well as the column name stored in generated segments.<br><br>Note that you can use a [`transformSpec`](#transformspec) if you want to rename columns during ingestion time. | none (required) |
 | createBitmapIndex | For `string` typed dimensions, whether or not bitmap indexes should be created for the column in generated segments. Creating a bitmap index requires more storage, but speeds up certain kinds of filtering (especially equality and prefix filtering). Only supported for `string` typed dimensions. | `true` |
+| multiValueHandling | Specify the type of handling for multi-value fields. Possible values are `sorted_array`, `sorted_set`, and `array`. `sorted_array` and `sorted_set` order the array upon ingestion. `sorted_set` removes duplicates. `array` ingests data as-is | `sorted_array` |
```
Comment (gianm) [inline docs/ingestion/index.md:483]: It'd be nice to link "multi-value fields" to the `../querying/multi-value-dimensions.md` doc, so these two pages cross-link to each other.

**Review by gianm — COMMENTED** [review]
I double-checked to make sure there are tests for this feature, and it looks like there are, so I think it's good to add the docs. (We wouldn't want to add docs for an untested feature!)

A couple of comments and then I think this is good to go.

**Comment by gianm** [comment]
> To someone more intune with Druid's inner workings: Do we really want alphabetical sorted to be the default for MV dimensions? Would it not be better to default to `ARRAY` in DimensionSchema L89?

Probably ARRAY or maybe SORTED_SET would be an ideal default. SORTED_ARRAY is a little weird, admittedly, but it's the default for legacy reasons. Maybe we'll change it in a future version.

**Comment by lkm** [comment]
Thanks @gianm. Requested changes added

**Review by sthetland — APPROVED** [review]
LGTM

**Pull request merged**